youtube_dl/extractor/generic.py

```diff
@@ -1817,6 +1817,11 @@ def filter_video(urls):
                     'url': new_url,
                 }
         if not found:
+            # Last resort: just look for anything that looks like a media URL
+            found = set(re.findall(
+                r'\b((?:https?:)?//[A-Za-z0-9.:[\]]+/[A-Za-z0-9\-\._~:/()*+,;=%]+\.(?:mp3|mp4|flv|m3u8|smil)(?:\?[A-Za-z0-9\-\._~:/[\]()*+,;=%]*)?)\b',
```

Comment:
I've yet cherry-picked your first commit in 12439dd5ec46ccac1ef13db7280473ecef44a096, thanks. As of second lets wait to hear other opinions.